src/form.js

```diff
@@ -145,29 +145,49 @@ define([
     }
 
     Form.prototype = {
-        addHashtag: function(hashtag, xpath) {
-            this.hashtagDictionary[hashtag] = xpath;
+        isValidHashtag: function(tag) {
+            return this.hashtagDictionary.hasOwnProperty(this.normalizeHashtag(tag));
+        },
+        addHashtag: function(hashtag, xpath, dontOverwrite) {
+            if (!dontOverwrite || !this.hashtagDictionary[hashtag]) {
+                this.hashtagDictionary[hashtag] = xpath;
+            }
         },
         removeHashtag: function(hashtag) {
             delete this.hashtagDictionary[hashtag];
         },
-        normalizeHashtag: function (xpath_) {
+        transform: function(input, transformFn) {
+            input = this.normalizeEscapedHashtag(input);
+            return escapedHashtags.transform(input, transformFn);
+        },
+        normalize: function (methodName, xpath) {
             // try catch is needed as workaround for having an itemset without
-            // the itemset plugin enabled
+            // the itemset plugin enabled and invalid xpaths
             try {
-                return xpath_ ? this.xpath.parse(xpath_).toHashtag() : xpath_;
+                return xpath ? this.xpath.parse(xpath)[methodName]() : xpath;
             } catch (err) {
-                return xpath_;
+                return xpath.startsWith('#invalid/xpath ') ? xpath.slice(15) : xpath;
```

Comment:
OK, not going to block on this, but I think it's important to revisit in a separate PR. I think it should be changed to something like
```js
} catch (err) {
    if (methodName !== "toEscapedHashtag") {
        throw err
    }
    return xpath.startsWith('#invalid/xpath ') ? xpath.slice(15) : xpath;
}
```
`toEscapedHashtag` is the only normalize method that knows or cares about the `#invalid/xpath ` prefix, and other methods should not be modifying the output or swallowing expressions that happen to have that prefix.